### PR TITLE
deps: support numpy 2.0

### PR DESCRIPTION
**Pull request by mloubout**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
 
 import os
+import sys
+import pkg_resources
 from setuptools import setup, find_packages
 
+
+def numpy_compat(required):
+    new_reqs = [r for r in required if "numpy" not in r and "sympy" not in r]
+    if sys.version_info < (3, 9):
```
Comment (georgebisbas) [inline setup.py:11]: (3, 9) ?

What is the format?

Comment (mloubout) [reply setup.py:11]: Thats the standard way to check python versions, this check <3.9

Comment (georgebisbas) [reply setup.py:11]: ok, looks weird

File: conftest.py [hunk 2]
```diff
@@ -453,3 +455,10 @@ def check_array(array, exp_halo, exp_shape, rotate=False):
 
     assert tuple(array.halo) == exp_halo
     assert tuple(shape) == tuple(exp_shape)
+
+
+# Make Ipython use str for numpy dtypes
```
Comment (georgebisbas) [inline conftest.py:460]: wow, is that necessary for numpy 2?

File: setup.py [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
+        # Numpy 2.0 requires python > 3.8
+        new_reqs.extend(["sympy>=1.9,<1.13", "numpy>1.16,<2.0"])
+        return new_reqs
+
+    # Due to api changes in numpy 2.0, it requires sympy 1.12.1 at the minimum
+    # Check if sympy is installed and enforce numpy version accordingly.
+    # If sympy isn't installed, endforce sympy>=1.12.1 and numpy>=2.0
```
Comment (georgebisbas) [inline setup.py:18]: "enforce"

**Review by FabioLuporini — APPROVED** [review]
Uncontroversial, thanks

**Review by EdCaunt — APPROVED** [review]

**Review by FabioLuporini — APPROVED** [review]

**Pull request merged**